ir/src/main/scala/org/scalajs/core/ir/Infos.scala

```diff
@@ -64,4 +70,292 @@ object Infos {
+    private val accessedModules = mutable.Set.empty[String]
+    private val accessedClassData = mutable.Set.empty[String]
+
+    def setEncodedName(encodedName: String): this.type = {
+      this.encodedName = encodedName
+      this
+    }
+
+    def setIsStatic(isStatic: Boolean): this.type = {
+      this.isStatic = isStatic
+      this
+    }
+
+    def setIsAbstract(isAbstract: Boolean): this.type = {
+      this.isAbstract = isAbstract
+      this
+    }
+
+    def setIsExported(isExported: Boolean): this.type = {
+      this.isExported = isExported
+      this
+    }
+
+    def addMethodCalled(receiverTpe: Type, method: String): this.type = {
+      receiverTpe match {
+        case ClassType(cls) => addMethodCalled(cls, method)
+        case AnyType        => addMethodCalled(ObjectClass, method)
+        case UndefType      => addMethodCalled(BoxedUnitClass, method)
+        case BooleanType    => addMethodCalled(BoxedBooleanClass, method)
+        case IntType        => addMethodCalled(BoxedIntegerClass, method)
+        case LongType       => addMethodCalled(BoxedLongClass, method)
+        case FloatType      => addMethodCalled(BoxedFloatClass, method)
+        case DoubleType     => addMethodCalled(BoxedDoubleClass, method)
+        case StringType     => addMethodCalled(StringClass, method)
+
+        case ArrayType(_, _) if method != "clone__O" =>
+          /* clone__O is overridden in the pseudo Array classes and is
+           * always kept anyway, because it is in scalajsenv.js.
+           * Other methods delegate to Object, which we can model with
+           * a static call to Object.method.
+           */
+          addMethodCalledStatically(ObjectClass, method)
+
+        case _ =>
+          // Nothing to do
+      }
+
+      this
+    }
+
+    def addMethodCalled(cls: String, method: String): this.type = {
+      methodsCalled.getOrElseUpdate(cls, mutable.Set.empty) += method
+      this
+    }
+
+    def addMethodCalledStatically(cls: String, method: String): this.type = {
+      methodsCalledStatically.getOrElseUpdate(cls, mutable.Set.empty) += method
+      this
+    }
+
+    def addStaticMethodCalled(cls: String, method: String): this.type = {
+      staticMethodsCalled.getOrElseUpdate(cls, mutable.Set.empty) += method
+      this
+    }
+
+    def addInstantiatedClass(cls: String): this.type = {
+      instantiatedClasses += cls
+      this
+    }
+
+    def addInstantiatedClass(cls: String, ctor: String): this.type =
+      addInstantiatedClass(cls).addMethodCalledStatically(cls, ctor)
+
+    def addAccessedModule(cls: String): this.type = {
+      accessedModules += cls
+      this
+    }
+
+    def addAccessedClassData(tpe: ReferenceType): this.type = {
+      addAccessedClassData(tpe match {
+        case ClassType(name)    => name
+        case ArrayType(base, _) => base
+      })
+    }
+
+    def addAccessedClassData(cls: String): this.type = {
+      if (!AlwaysPresentClassData.contains(cls))
```

Comment:
Well it's just part of the spec of the info that this precise set of classes is never mentioned. The set is not arbitrary: it is tightly coupled with the semantics of Scala.js. There are other class data never eliminated by the linker that are not mentioned here, because those are artifacts of the linker. But `Object`, primitive classes and hijacked classes must necessarily always be present due to the semantics.

An alternative, if you still don't like it, is to exclude only `PrimitiveClasses` here (the `Analyzer` would not be happy if it received an `I` here).